module/modmanager/manager.go

```diff
@@ -1275,15 +1319,24 @@ func (m *module) cleanupAfterCrash(mgr *Manager) {
 }
 
 func (m *module) getFullEnvironment(viamHomeDir string) map[string]string {
+	return getFullEnvironment(m.cfg, m.dataDir, viamHomeDir)
+}
+
+// TODO: consider making this a module config method.
+func getFullEnvironment(
+	cfg config.Module,
+	dataDir string,
+	viamHomeDir string,
+) map[string]string {
 	environment := map[string]string{
 		"VIAM_HOME":        viamHomeDir,
-		"VIAM_MODULE_DATA": m.dataDir,
+		"VIAM_MODULE_DATA": dataDir,
 	}
-	if m.cfg.Type == config.ModuleTypeRegistry {
-		environment["VIAM_MODULE_ID"] = m.cfg.ModuleID
+	if cfg.Type == config.ModuleTypeRegistry {
+		environment["VIAM_MODULE_ID"] = cfg.ModuleID
 	}
 	// Overwrite the base environment variables with the module's environment variables (if specified)
-	for key, value := range m.cfg.Environment {
+	for key, value := range cfg.Environment {
 		environment[key] = value
 	}
 	return environment
```

Comment:
refactored to make this functionality available to `FirstRun` more conveniently.